### PR TITLE
SH-21: Fix the logic of filtering

### PR DESCRIPTION
**Pull request by WeszeRita**
## JIRA ticket link:
https://shoperz.atlassian.net/browse/SH-21

### CODE REVIEW

File: src/shared/services/filter.service.ts [hunk 1]
```diff
@@ -5,20 +5,20 @@ import {
   ICalculatedProduct,
   IFilterOption,
   IFilterDefinition,
-} from '../models';
-import { ProductFilterOption } from '../enums';
-import { CategoryFacadeService, BrandFacadeService, FilterFacadeService } from '../services';
+  ProductFilterOption,
+  CategoryFacadeService,
+  BrandFacadeService,
+} from '@shared-module';
```
Comment (krucsodaniel) [inline src/shared/services/filter.service.ts]: Why did you change the imports here? It should remain relative and not use the alias.

File: src/shared/services/filter.service.ts [hunk 2]
```diff
@@ -5,20 +5,20 @@ import {
...
 import { TranslateService } from '@ngx-translate/core';
 import { firstValueFrom } from 'rxjs';
 import { Store } from '@ngrx/store';
-import { FilterActions } from '../store';
+import { FilterActions } from '../store/filters';
```
Comment (krucsodaniel) [inline src/shared/services/filter.service.ts]: Why did you change the imports here? It was correct as it was before, because we are using index files for importing.

**Review by krucsodaniel — CHANGES_REQUESTED** [review]

**Review by krucsodaniel — APPROVED** [review]

**Review by bodnya29179 — APPROVED** [review]
